src/profile-logic/process-profile.js

```diff
@@ -1052,56 +1056,195 @@ export function processProfile(
 }
 
 /**
- * Take a processed profile and remove any non-serializable classes such as the
- * StringTable class.
+ * Take a processed profile with PII that will be removed and remove any
```

Comment:
I find myself struggling understanding this function. It seems to be combining two steps.

1. Serializing profiles
2. Removing PII.

It seems like it would be simpler to have 2 different functions here that are composed to do the correct thing. It might be a few more lines of code, but I think it would be simpler to understand.